pl.c

```diff
@@ -459,6 +452,38 @@ static void pl_delete_selected_pl(void)
 	job_schedule_pl_delete(pdd);
 }
 
+static void pl_delete_selected_pl(void)
+{
+	if (yes_no_query("Delete selected playlist? [y/N]") != UI_QUERY_ANSWER_YES)
+		return;
+
+	pl_delete(pl_visible);
+}
+
+void pl_delete_by_name(char *name)
+{
+	struct playlist *pl;
+	list_for_each_entry(pl, &pl_head, node) {
+		if (strcmp(pl->name, name) == 0) {
+			pl_delete(pl);
+			return;
+		}
+	}
+	error_msg("couldn't find a playlist named '%s' to delete", name);
+}
+
+void pl_delete_all(void)
+{
+	struct playlist *pl;
+	struct playlist *temp;
+	list_for_each_entry_safe(pl, temp, &pl_head, node) {
+		if (list_len(&pl_head) == 1)
+			break;
```

Comment:
Still not getting this, won't the for_each loop just end naturally when all the items are gone?

Reply:
Wait I get it, because we add back the default :grin:

Reply:
Yeah it could still be fine but I couldn't really wrap my head around the implementation too well to confirm it.